packages/openactive-integration-tests/test/features/approval/minimal-proposal/implemented/accept-proposal-book-test.js

```diff
@@ -42,50 +42,351 @@ FeatureHelper.describeFeature(module, {
+  //     createActionFn: () => ({
+  //       type: 'test:SellerAcceptOrderProposalSimulateAction',
+  //       objectType: 'OrderProposal',
+  //       objectId: p.getResponse().body['@id'],
+  //     }),
+  //   })],
+  //   ['orderFeedUpdate', 'OrderFeedUpdate', {
+  //     testName: 'Order Feed Update (after Simulate Seller Approval)',
+  //   }],
+  //   ['b', BFlowStage.create],
+  // ], { logger, requestHelper });
+
+  // # Examples
+  //
+  // ## 1. Simple
+  // ## 1.1. With no magic at all
+  {
+    const uuid = createUuid();
+    const sellerId = getSellerId();
+    const fetchOpportunities = FetchOpportunitiesFlowStage.create({
+      orderItemCriteriaList,
+      requestHelper,
+      logger,
+    });
+    const c1 = C1FlowStage.create({
+      getInputState: () => ({
+        orderItems: fetchOpportunities.getOutput().orderItems,
+      }),
+      prerequisite: fetchOpportunities,
+      uuid,
+      sellerId,
+      requestHelper,
+      logger,
+    });
+    const c2 = C2FlowStage.create({
+      getInputState: () => ({
+        orderItems: fetchOpportunities.getOutput().orderItems,
+      }),
+      prerequisite: c1,
+      uuid,
+      sellerId,
+      requestHelper,
+      logger,
+    });
+    const p = PFlowStage.create({
+      getInputState: () => ({
+        orderItems: fetchOpportunities.getOutput().orderItems,
+        totalPaymentDue: c2.getOutput().totalPaymentDue,
+      }),
+      prerequisite: c2,
+      uuid,
+      sellerId,
+      requestHelper,
+      logger,
+    });
+    const listenForOrderFeedUpdate = OrderFeedUpdateFlowStage.createListener({
+      prerequisite: p,
+      uuid,
+      requestHelper,
+    });
+    const simulateSellerApproval = TestInterfaceActionFlowStage.create({
+      testName: 'Simulate Seller Approval (Test Interface Action)',
+      prerequisite: listenForOrderFeedUpdate,
+      createActionFn: () => ({
+        type: 'test:SellerAcceptOrderProposalSimulateAction',
+        objectType: 'OrderProposal',
+        objectId: p.getOutput().orderId,
+      }),
+      requestHelper,
+    });
+    const collectOrderFeedUpdate = OrderFeedUpdateFlowStage.createCollector({
+      getInputState: () => ({
+        getOrderFromOrderFeedPromise: listenForOrderFeedUpdate.getOutput().getOrderFromOrderFeedPromise,
+      }),
+      testName: 'Order Feed Update (after Simulate Seller Approval)',
+      prerequisite: simulateSellerApproval,
+      logger,
+    });
+    const b = BFlowStage.create({
+      getInputState: () => ({
+        orderItems: fetchOpportunities.getOutput().orderItems,
+        orderProposalVersion: p.getOutput().orderProposalVersion, // (this could also use collectOrderFeedUpdate.getOutput())
+        totalPaymentDue: p.getOutput().totalPaymentDue,
+      }),
+      prerequisite: collectOrderFeedUpdate,
+      logger,
+      requestHelper,
+    });
+  }
+  // ## 1.2. With simple substitutions / helper functions
```

Comment:
Am still skeptical of using merging, even via the helpers, as we could get inconsistent behaviour this way... I wonder if there are other options here?